src/ansible_navigator/configuration_subsystem/definitions.py

```diff
@@ -134,16 +138,22 @@ class SubCommand(SimpleNamespace):
     epilog: Union[None, str] = None
 
 
-class ApplicationConfiguration(SimpleNamespace):
+@dataclass
+class ApplicationConfiguration:
+    # pylint: disable=too-many-instance-attributes
     """The main object for storing an application config"""
 
-    application_name: str = ""
-    entries: List[SettingsEntry] = []
+    application_version: str
+    entries: List[SettingsEntry]
     internals: SimpleNamespace
+    post_processor: "NavigatorPostProcessor"
     subcommands: List[SubCommand]
-    post_processor = Callable
+
+    application_name: str = ""
+    #  = field(default_factory=list)
```

Comment:
nuke

Reply:
_(could've been caught by `eradicate`)_

Reply:
I think that would be a great one to try out.  If you help out with the docstrings, we can do that next....

Reply:
I don't think it will need any effort beyond reverting the removal you did.